pkg/diagnose/connectivity/core_endpoint_test.go

```diff
@@ -59,3 +60,47 @@ func TestSendHTTPRequestToEndpoint(t *testing.T) {
 	assert.Equal(t, statusCode, 400)
 	assert.Equal(t, string(responseBody), "Bad Request")
 }
+
+func TestAcceptRedirection(t *testing.T) {
+	testCases := []struct {
+		name        string
+		handlerFunc http.HandlerFunc
+		fail        bool
+	}{
+		{
+			name: "ok",
+			handlerFunc: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				//  * the original flare request URL, which redirects on HEAD to /post-target
+				if r.Method == "HEAD" && r.RequestURI == "/support/flare" {
+					// redirect to /post-target.
+					w.WriteHeader(307)
+				} else {
+					w.WriteHeader(500)
+					io.WriteString(w, "path not recognized by httptest server")
+				}
+			}),
+			fail: false,
+		},
+	}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏No need to declare `testCases` and loop over it if you have a single test case, you could just inline it 😄

Reply:
fixed